src/geometry.rs

```diff
@@ -447,86 +361,186 @@ impl Hash for UnorientedEdge {
     }
 }
 
-#[allow(dead_code, clippy::type_complexity)]
 fn remove_orphan_vertices(
-    faces: Vec<(u32, u32, u32)>,
-    vertices: Vec<Point3<f32>>,
-) -> (Vec<(u32, u32, u32)>, Vec<Point3<f32>>) {
-    let mut old_to_new_vertex_index: HashMap<u32, u32> = HashMap::new();
-    let mut reduced_vertices: Vec<Point3<f32>> = Vec::new();
-    let mut new_index = 0;
-    #[allow(clippy::needless_range_loop)]
-    for original_index in 0..vertices.len() {
-        let o_i = cast_u32(original_index);
-        if faces.iter().any(|f| f.0 == o_i || f.1 == o_i || f.2 == o_i) {
-            old_to_new_vertex_index.insert(o_i, new_index);
-            reduced_vertices.push(vertices[original_index]);
-            new_index += 1;
-        }
-    }
-
-    let recomputed_faces: Vec<_> = faces
-        .iter()
-        .map(|f| {
-            (
-                *old_to_new_vertex_index
-                    .get(&f.0)
-                    .expect("Vertex index not found"),
-                *old_to_new_vertex_index
-                    .get(&f.1)
-                    .expect("Vertex index not found"),
-                *old_to_new_vertex_index
-                    .get(&f.2)
-                    .expect("Vertex index not found"),
-            )
-        })
-        .collect();
-
-    (recomputed_faces, reduced_vertices)
+    faces: FaceVertexTuples,
+    vertices: Vertices,
+) -> (FaceVertexTuples, Vertices) {
+    let mut vertices_reduced: Vertices = Vec::new();
```

Comment:
Let's create some sensible breathing space for this function (and the normals-only one) and insert a few newlines here and there. The style in the vertex and normal orphan removal function is good to apply here too.